pkg/kubehound/graph/builder.go

```diff
@@ -142,6 +163,17 @@ func (b *Builder) buildDependent(ctx context.Context, l *log.KubehoundLogger, oi
 	for label, e := range b.edges.Dependent() {
 		err := b.buildEdge(ctx, label, e, oic, l)
 		if err != nil {
+			// In case we don't want to continue and have a partial graph built, we return an error.
+			// This then fails the WaitForComplete early and bubbles up to main.
+			if b.cfg.Builder.StopOnError {
+				return err
+			}
+			// Otherwise, by returning nil AND printing an error, we explicitly tell the user that something is going wrong.
+			// Since the issue might not be easy or even possible for the user to fix, we still want to be able to provide _some_
+			// values to the user (permissions of the users etc...)
+			// TODO(#ASENG-512): Add an error handling framework to accumulate all errors and display them to the user in an user friendly way
+			l.Errorf("Failed to create a dependent edge (type: %s). The created graph will be INCOMPLETE (change `builder.stop_on_error` to abort or error instead)", e.Name())
+
 			return fmt.Errorf("building dependent edge %s: %w", label, err)
```

Comment:
The comment says we want to return nil here, but I believe we are still returning an error (Just started golang so I might overlook smthg, please tell me if we actually return nil and I don't know the pattern 😄)

Reply:
ooh...
Good catch :sweat_smile:. I shouldn't code late
Fixed in both place thanks!